rq/job.py

```diff
@@ -676,27 +678,47 @@ def save_meta(self):
         meta = self.serializer.dumps(self.meta)
         self.connection.hset(self.key, 'meta', meta)
 
-    def cancel(self, pipeline=None):
+    def cancel(self, pipeline=None, enqueue_dependents=False):
         """Cancels the given job, which will prevent the job from ever being
         ran (or inspected).
 
         This method merely exists as a high-level API call to cancel jobs
         without worrying about the internals required to implement job
         cancellation.
-        """
-        pipeline = pipeline or self.connection.pipeline()
-        if self.origin:
-            from .registry import CanceledJobRegistry
-            from .queue import Queue
 
-            q = Queue(name=self.origin, connection=self.connection)
-            q.remove(self, pipeline=pipeline)
-
-            self.set_status(JobStatus.CANCELED, pipeline=pipeline)
-
-            registry = CanceledJobRegistry(self.origin, self.connection, job_class=self.__class__)
-            registry.add(self, pipeline=pipeline)
-        pipeline.execute()
+        You can enqueue the jobs dependents optionally, 
+        Same pipelining behavior as Queue.enqueue_dependents on whether or not a pipeline is passed in.
+        """
+        pipe = pipeline or self.connection.pipeline()
+        while True:
+            try:
+                if self.origin:
```

Comment:
We can assume `self.origin` is always there. This check was added a long time ago when the `origin` property was newly added.

Reply:
@selwin You want me to remove tis check?